contrib/roles/windows/kubernetes/tasks/run_minion_init.yml

```diff
@@ -19,6 +19,36 @@
   win_path:
     elements: "{{ install_info.install_path }}"
 
+- name: Kubernetes minion | check if ovn-kubernetes-node is installed
+  win_service:
+    name: ovn-kubernetes-node
+  register: ovnkube_service
+
+- name: Kubernetes minion | Remove ovn-kubernetes-node service if it exists
+  block:
+    - name: Kubernetes minion | Stop ovn-kubernetes-node service
+      win_service:
+        name: ovn-kubernetes-node
+        state: stopped
+      changed_when: false
+
+    - name: Kubernetes minion | Remove ovn-kubernetes-node service
+      win_service:
+        name: ovn-kubernetes-node
+        state: absent
+      changed_when: false
+  when: ovnkube_service.exists
+
+- name: Kubelet | Create ovn-kubernetes-node service
+  win_service:
+    name: ovn-kubernetes-node
+    path: '"{{ install_info.install_path }}\\servicewrapper.exe" ovn-kubernetes-node {{ install_info.install_path }}\\ovnkube.exe --k8s-kubeconfig={{ install_info.install_path }}\\kubeconfig.yaml --k8s-apiserver http://{{ kubernetes_info.MASTER_IP }}:8080 --init-node {{ ansible_hostname|lower }} --k8s-token {{TOKEN}} --nb-address "tcp://{{ kubernetes_info.MASTER_IP }}:6641" --sb-address "tcp://{{ kubernetes_info.MASTER_IP }}:6642" --cluster-subnet {{ kubernetes_info.CLUSTER_SUBNET }} --cni-conf-dir="{{ install_info.install_path }}/cni" --cni-plugin "ovn-k8s-cni-overlay.exe"'
```

Comment:
Please add the following argument ``--encap-ip={{ host_internal_ip }}`` to ovnkube.

Reply:
I don't really like this approach because the adapter may change(i.e. DHCP server) . We need to improve the IP detection in the end IMO.
I will add it as a workaround and open an issue for the IP detection.